### PR TITLE
Add guide to raise a patch in Lighthouse.

### PR DESCRIPTION
**Pull request by aswinsuryan**
Signed-off-by: Aswin Surayanarayanan <asuryana@redhat.com>

### CODE REVIEW

File: src/content/contributing/lighthouse/_index.en.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+---
```
Comment (mangelajo) [inline src/content/contributing/lighthouse/_index.en.md:1]: @aswinsuryan please use make server locally in your laptop to verify the rendering, it's not working in a few places (see comments bellow).

File: src/content/contributing/lighthouse/_index.en.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+Once you run the make e2e you will have a 3 virtual cluster deployment, you could use it to verify if the changes are working as expected manually. Lighthouse agent will be running the submariner-operator namespace in the data clusters. You could use kubectl commands to see the logs and the pods status.
+
+####Contrubuting to Lighthouse Plugin
+
+The plugin/lighthouse/ folder has the lighthouse plugin code. This uses the structure of as required by a standard code DNS plugin. The plugin answers the DNS requests by using the info in multicluster service. After making changes you can [build](https://submariner-io.github.io/contributing/building_testing/#submariner-iolighthouse) and [test](https://submariner-io.github.io/contributing/building_testing/#common-build-and-testing-targets) the code.
```
Comment (mangelajo) [inline src/content/contributing/lighthouse/_index.en.md:27]:
```suggestion
The plugin/lighthouse/ folder has the lighthouse plugin code. This uses the structure of as required by a standard code DNS plugin. The plugin answers the DNS requests by using the info in multicluster service. After making changes you can [build](../building_testing/#submariner-iolighthouse) and [test](.../building_testing/#common-build-and-testing-targets) the code.
```

File: src/content/contributing/lighthouse/_index.en.md (hunk 1 above)
Comment (aswinsuryan) [reply src/content/contributing/lighthouse/_index.en.md:1]: Oh I missed that step for this one, thanks for checking it. Will verify it.

**Comment by aswinsuryan** [comment]
Closes https://github.com/submariner-io/submariner-website/issues/117

File: src/content/contributing/lighthouse/_index.en.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
+---
+title: "Contributing to Lighthouse"
```
[the remark this answers is not in the document]
Comment (aswinsuryan) [reply src/content/contributing/lighthouse/_index.en.md:2]: Added line wraps.

File: src/content/contributing/lighthouse/_index.en.md (hunk 1 above)
Comment (aswinsuryan) [reply src/content/contributing/lighthouse/_index.en.md:1]: Fixed the issues.

File: src/content/contributing/lighthouse/_index.en.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+git clone https://github.com/<your-github-username>/lighthouse.git
+cd lighthouse
+git checkout -b <BRANCH-NAME>
+```
+
+#### Contributing to Lighthouse Agent
+
+The Lighthouse agent code is present at pkg folder inside the Lighthouse house directory. There is a package for Kubernetes controller
+which listens for service creation and creates MultiClusterService CRs. Once you have made the code changes you can refer [building](../building_testing/#submariner-iolighthouse)
+and [testing](../building_testing/#common-build-and-testing-targets) it to check if the changes have no regression.
```
Comment (tpantelis) [inline src/content/contributing/lighthouse/_index.en.md:26]:
```suggestion
The Lighthouse Agent code is located under the `pkg/agent` directory. The `Controller` is the main component that performs bi-directional syncing of Service and MultiClusterService resources. After making code changes, please refer to [building](../building_testing/#submariner-iolighthouse)
and [testing](../building_testing/#common-build-and-testing-targets) to build and verify your changes.
```

File: src/content/contributing/lighthouse/_index.en.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+Once you run the make e2e you will have a 3 virtual cluster deployment, you could use it to verify if the changes are working as expected manually.
+Lighthouse agent will be running the submariner-operator namespace in the data clusters. You could use kubectl commands
+to see the logs and the pods status.
```
Comment (tpantelis) [inline src/content/contributing/lighthouse/_index.en.md:30]:
```suggestion
Running `make e2e` will create a 3-cluster deployment which you can use it manually verify changes if necessary.
The Lighthouse Agent will be deployed in the `submariner-operator` namespace in each cluster. You can use `kubectl` commands to check the pod's logs and status.
```

File: src/content/contributing/lighthouse/_index.en.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+#### Contributing to Lighthouse Plugin
+
+The plugin/lighthouse/ folder has the lighthouse plugin code. This uses the structure of as required by a standard code DNS plugin.
+The plugin answers the DNS requests by using the info in multicluster service. After making changes you can [build](../building_testing/#submariner-iolighthouse)
+and [test](.../building_testing/#common-build-and-testing-targets) the code.
```
Comment (tpantelis) [inline src/content/contributing/lighthouse/_index.en.md:36]:
```suggestion
The Lighthouse plugin code is located under the `plugin/lighthouse` directory and uses the structure as required by a standard code CoreDNS plugin.
The plugin answers the DNS requests by using the info in multicluster service. After making code changes, please refer to [building](../building_testing/#submariner-iolighthouse)
and [testing](../building_testing/#common-build-and-testing-targets) to build and verify your changes.
```

File: src/content/contributing/lighthouse/_index.en.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+Similar to the agent, you can use the deployment created using make e2e. The lighthouse plugin will be running as a part of the coreDNS in the kube-system 
+namespace and the config map will be changed to use it. The usual kubectl commands can be used here as well.
+You can add the debug entry in the coreDNS config map if you require more logging.
```
Comment (tpantelis) [inline src/content/contributing/lighthouse/_index.en.md:40]:
```suggestion
Similar to the agent, you can use the deployment created using `make e2e` to manually test and verify as well. The Lighthouse plugin will be running as a part of the CoreDNS in the `kube-system`  
namespace and the CoreDNS config map will be changed to use it. The usual `kubectl` commands can be used here as well.
You can add the debug entry in the CoreDNS config map if you require more logging.
```

File: src/content/contributing/lighthouse/_index.en.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+### Raising a Pull Request
```
Comment (tpantelis) [inline src/content/contributing/lighthouse/_index.en.md:42]:
```suggestion
### Creating a Pull Request
```

File: src/content/contributing/lighthouse/_index.en.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+You can commit your changes using git commit and push the changes to your fork branch. Then you will see an option to raise a PR in
+the GitHub UI in your browser when you go to your forked repo.
```
Comment (tpantelis) [inline src/content/contributing/lighthouse/_index.en.md:45]:
```suggestion
You can commit your changes using `git commit` and push the changes to your fork branch. Then you will see an option to create a PR in
the GitHub UI in your browser when you go to your forked repository.
```

File: src/content/contributing/lighthouse/_index.en.md [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+
+```
+git commit -s #Give a commit message
+git push -f origin HEAD:<Your-Branch-Name>
+```
+
+Raising a PR will trigger CI and will verify the changes.
```
Comment (tpantelis) [inline src/content/contributing/lighthouse/_index.en.md:52]:
```suggestion
Creating a PR will trigger CI to verify the changes.
```

File: src/content/contributing/lighthouse/_index.en.md [hunk 11]
```diff
@@ -0,0 +1,52 @@
+---
+title: "Contributing to Lighthouse"
+date: 2020-04-20T17:12:52+0300
+weight: 10
+---
+
+The [Lighthouse Project](https://github.com/submariner-io/lighthouse) has an agent and a coreDNS plugin.
+You can find more about the architecture at [Lighthouse Architecture](../../architecture/components/lighthouse/).
+This guide helps in making your contribution easier.
```
[the remark this answers is not in the document]
Comment (aswinsuryan) [reply src/content/contributing/lighthouse/_index.en.md]: Done.

File: src/content/contributing/lighthouse/_index.en.md [hunk 12]
```diff
@@ -21,32 +20,30 @@ git checkout -b <BRANCH-NAME>
...
-which listens for service creation and creates MultiClusterService CRs. Once you have made the code changes you can refer [building](../building_testing/#submariner-iolighthouse)
-and [testing](../building_testing/#common-build-and-testing-targets) it to check if the changes have no regression.
+The Lighthouse Agent code is located under the `pkg/agent` directory. The `Controller` is the main component that performs bi-directional syncing of Service and MultiClusterService resources. After making code changes, please refer to [building](../building_testing/#submariner-iolighthouse)
+and [testing](../building_testing/#common-build-and-testing-targets) to build and verify your changes.
 
-Once you run the make e2e you will have a 3 virtual cluster deployment, you could use it to verify if the changes are working as expected manually.
-Lighthouse agent will be running the submariner-operator namespace in the data clusters. You could use kubectl commands
-to see the logs and the pods status.
+Running `make e2e` will create a 3-cluster deployment which you can use it manually verify changes if necessary.
```
Comment (tpantelis) [inline src/content/contributing/lighthouse/_index.en.md:26]:
```suggestion
Running `make e2e` will create a 3-cluster deployment which you can use to manually verify changes if necessary.
```

File: src/content/contributing/lighthouse/_index.en.md [hunk 13]
```diff
@@ -21,32 +20,30 @@ git checkout -b <BRANCH-NAME>
...
+The Lighthouse Agent will be deployed in the `submariner-operator` namespace in each cluster. You can use `kubectl` commands to check the pod's logs and status.
 
 #### Contributing to Lighthouse Plugin
 
-The plugin/lighthouse/ folder has the lighthouse plugin code. This uses the structure of as required by a standard code DNS plugin.
-The plugin answers the DNS requests by using the info in multicluster service. After making changes you can [build](../building_testing/#submariner-iolighthouse)
-and [test](.../building_testing/#common-build-and-testing-targets) the code.
+The Lighthouse plugin code is located under the `plugin/lighthouse` directory and uses the structure as required by a standard code CoreDNS plugin.
```
Comment (tpantelis) [inline src/content/contributing/lighthouse/_index.en.md:31]:
```suggestion
The Lighthouse plugin code is located under the `plugin/lighthouse` directory and uses the structure as required by a standard CoreDNS plugin.
```

File: src/content/contributing/lighthouse/_index.en.md [hunk 14]
```diff
@@ -21,32 +20,30 @@ git checkout -b <BRANCH-NAME>
...
+The plugin answers the DNS requests by using the info in multicluster service. After making code changes, please refer to [building](../building_testing/#submariner-iolighthouse)
```
Comment (tpantelis) [inline src/content/contributing/lighthouse/_index.en.md:32]:
```suggestion
The plugin answers DNS requests by using the MultiClusterService resources populated by the Lighthouse Agent. After making code changes, please refer to [building](../building_testing/#submariner-iolighthouse)
```

**Comment by mangelajo** [comment]
@aswinsuryan you have some comments by tom before we can merge.

**Pull request merged**